code/modules/antagonists/swarmer/swarmer.dm

```diff
@@ -364,6 +364,34 @@
 			return TRUE
 	return ..()
 
+/obj/structure/foamedmetal/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	var/isonshuttle = istype(loc, /area/shuttle)
+	for(var/turf/T in range(1, src))
+		var/area/A = get_area(T)
+		if(isspaceturf(T) || (!isonshuttle && (istype(A, /area/shuttle) || istype(A, /area/space))) || (isonshuttle && !istype(A, /area/shuttle)))
+			to_chat(S, "<span class='warning'>Destroying this object has the potential to cause a hull breach. Aborting.</span>")
+			S.target = null
+			return TRUE
+		else if(istype(A, /area/engine/supermatter))
+			to_chat(S, "<span class='warning'>Disrupting the containment of a supermatter crystal would not be to our benefit. Aborting.</span>")
+			S.target = null
+			return TRUE
+	return ..()
+
+/obj/structure/foamedmetal/resin/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
```

Comment:
am I missing something, what's different between `foamedmetal` and `foamedmetal/resin`?

Reply:
Foamed metal is metal foam walls, resin is atmos resin from the atmos backpack.

Reply:
I meant to ask why is resin not inheriting

Reply:
For some reason I thought it wasn't inheriting, turns out it does. I'll remove the redundant part